### DISCUSSION TITLE
Is there a way to force a scope to be provided?

### DISCUSSION THREAD

**Discussion by scamden** [saasquatch/bunshi#56] [Q&A] [answered]
My use case. I have a custom bunshi scope. I want to ensure when the molecules that depend on that scope are used, they are always used inside a ScopeProvider. 

In other words, I'd like to disable the global fallback for scope so that consumers of this molekule don't accidentally start using their state globally and interact with each other. 

is this possible?

**Comment by scamden** [comment, marked as answer]
Answered my own question. 
```ts
const myScope = createScope<string | undefined>(undefined);
const myMolecule = molecule(() => {
  const scope = use(myScope);
  if (!scope) {
    throw new Error("scope is undefined");
  }
  const countAtom = atom(0);
  return {
    countAtom: countAtom,
  };
});
```

**Comment by scamden** [comment]
Small follow up question: any ideas on how to ensure that all molecules within a given file folder use a scope? I want to make sure all the state within my folder's subtree is scoped to one particular custom scope and I feel like it will be easy for developers coming after me to create molecules and forget to add the `use(myScope)`. Maybe a lint rule? any other ideas?

**Reply by loganvolkers** [reply]
Enforcing custom rules in JavaScript files based on which folder they're in sounds like a job for Grit https://www.grit.io/

I have no experience with it, but it's definitely beyond the scope of bunshi to enforce these kinds of rules based on folder. 

If you're writing a lot of these molecules, then you might want to create some functions to reduce the boilerplate, like a `function customScopedMolecule` that internally creates and returns a molecule.

**Reply by scamden** [reply]
Grit looks interesting! Thanks for that! 

> but it's definitely beyond the scope of bunshi to enforce these kinds of rules based on folder

Of course :) I was just wondering if anyone else had experience with using them at a large company and had ideas or best practices. I had written a custom eslint rule for jotai-scope that ensured we used our custom scoping function and it seems like it might even be easier with this but always curious what others are doing :)
